src/util.h

```diff
@@ -278,6 +289,276 @@ writable_buffer_append(writable_buffer *wb, uint64 length, const void *newdata)
    writable_buffer_copy_slice(&dst##wb, src);                                  \
    slice dst = writable_buffer_to_slice(&dst##wb);
 
+/*
+ * ----------------------------------------------------------------------
+ * Fingerprint Header: Fingerprints are managed while building routing filters.
```

Comment:
This almost wants to be in its own file: `fingerprint.h` and included in the two consumers: `btree.c` and `trunk.c`.

Wanted to ask your opinions before adding a new file.